src/sage/schemes/elliptic_curves/period_lattice.py

```diff
@@ -231,14 +232,14 @@ def __init__(self, E, embedding=None):
         # the given embedding:
 
         K = E.base_field()
-        self.is_approximate = isinstance(K, (RealField_class, ComplexField_class))
+        self._is_exact = K.is_exact()
         if embedding is None:
```

Comment:
Since `is_approximate` is in public API, you need to add
```python
self.is_approximate = self._is_exact
```
unless you plan to deprecate it.

Reply:
Evidently naming the thing without an underscore was a bad idea… but sure.

Reply:
Hmm. We cannot regard (mechanically) a class variable without an underscore as in public API. But I thought `is_approximate` was intended so (to be used by a user). This may be my mistake...

If you don't agree, then I would respect your judgment. You know more about the class than I do.

Reply:
Already add it back as deprecated property.